src/go/rdctl/cmd/factoryReset.go

```diff
@@ -40,23 +48,264 @@ Use the --remove-kubernetes-cache=BOOLEAN flag to also remove the cached Kuberne
+	}
+	configPath := path.Join(configHomePath, "rancher-desktop")
+
+	cacheHomePath := os.Getenv("XDG_CACHE_HOME")
+	if cacheHomePath == "" {
+		cacheHomePath = path.Join(homeDir, ".cache")
+	}
+	cachePath := path.Join(cacheHomePath, "rancher-desktop")
+
+	altAppHomePath := path.Join(homeDir, ".rd")
+
+	pathList := []string{
+		altAppHomePath,
+		configPath,
+		path.Join(configHomePath, "Rancher Desktop"),
+		path.Join(homeDir, ".local", "state", "rancher-desktop"),
+	}
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath != "" {
+		pathList = append(pathList, logsPath, path.Join(dataHomePath, "lima"))
+	} else {
+		pathList = append(pathList, path.Join(dataHomePath))
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteLinuxLikeData(altAppHomePath, configHomePath, pathList)
+}
+
+// XXX: Windows doesn't have symlinks, do we have to clean up any cli-plugins?
+func deleteWindowsData() error {
+	err := factoryreset.UnregisterWSL()
+	if err != nil {
+		return err
+	}
+	err = factoryreset.DeleteWindowsData(!removeKubernetesCache, "rancher-desktop")
+	if err != nil {
+		return err
+	}
+	return clearDockerContext()
+}
+
+func deleteLinuxLikeData(altAppHomePath string, configHomePath string, pathList []string) error {
+	deleteLimaVM()
+
+	for _, currentPath := range pathList {
+		err := os.RemoveAll(currentPath)
+		if err != nil {
+			return fmt.Errorf("error removing %s: %w", currentPath, err)
+		}
+	}
+	err := clearDockerContext()
+	if err != nil {
+		return err
+	}
+	err = removeDockerCliPlugins(altAppHomePath)
+	if err != nil {
+		return err
+	}
+	homeDir := os.Getenv("HOME")
+	rawPaths := []string{
+		".bashrc",
+		".bash_profile",
+		".bash_login",
+		".profile",
+		".zshrc",
+		".cshrc",
+		".tshrc",
+	}
+	for i, s := range rawPaths {
+		rawPaths[i] = path.Join(homeDir, s)
+	}
+	rawPaths = append(rawPaths, path.Join(configHomePath, "fish", "config.fish"))
+
+	return removePathManagement(rawPaths)
+}
+
+func deleteLimaVM() error {
+	err := setupLimaHome()
+	if err != nil {
+		return err
+	}
+	execPath, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	execPath, err = filepath.EvalSymlinks(execPath)
+	if err != nil {
+		return err
+	}
+	limactl := path.Join(path.Dir(path.Dir(execPath)), "lima", "bin", "limactl")
+	return exec.Command(limactl, "delete", "-f", "0").Run()
+}
+
+func removeDockerCliPlugins(altAppHomePath string) error {
+	cliPluginsDir := path.Join(dockerconfig.Dir(), "cli-plugins")
+	entries, err := os.ReadDir(cliPluginsDir)
+	if err != nil {
+		if errors.Is(err, syscall.ENOENT) {
+			// Nothing left to do here, since there is no cli-plugins dir
+			return nil
+		}
+		return err
+	}
+	for _, entry := range entries {
+		if entry.Type()&os.ModeSymlink != os.ModeSymlink {
+			continue
+		}
+		fullPathName := path.Join(cliPluginsDir, entry.Name())
+		target, err := os.Readlink(fullPathName)
+		if err != nil {
+			continue
+		}
+		if strings.Contains(target, altAppHomePath) {
```

Comment:
Wouldn't it be `strings.HasPrefix` instead? Of course, it really should be based off `filepath.Rel` to handle weird cases…

Reply:
This actually triggered a bunch of changes. Main one is to change the `Contains` line to

`strings.HasPrefix(target, path.Join(altAppHomePath, "bin")+"/")` (because using `filepath.Rel` strikes me as being too obtuse for this purpose).